int32 options
---
boolean setTime
---
boolean settingTime